carp.deployments.core/src/commonMain/kotlin/dk/cachet/carp/deployments/application/ParticipationServiceHost.kt

```diff
@@ -101,18 +101,18 @@ class ParticipationServiceHost(
     }
 
     /**
-     * Set participant [data] for the given [inputDataType] in the study deployment with [studyDeploymentId].
+     * Set participant [data] in the study deployment with [studyDeploymentId].
      *
      * @throws IllegalArgumentException when:
      *   - there is no study deployment with [studyDeploymentId]
-     *   - [inputDataType] is not configured as expected participant data in the study protocol
-     *   - [data] is invalid data for [inputDataType]
+     *   - one or more of the keys in [data] isn't configured as expected participant data in the study protocol
```

Comment:
Just a thought: should we really replicate documentation verbatim on the implementations?

Reply:
I've been wondering about this. I guess I need to check dokka/IntelliJ support.

I.e., does IntelliJ by default use the interface documentation if nothing is set on the concrete class? And, does dokka do the same when generating docs? I'll create an issue to look into this.

Reply:
https://github.com/cph-cachet/carp.core-kotlin/issues/304